[2023-12-21T16:52:56.706+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_ChestScan_with_mode scheduled__2023-12-01T00:00:00+00:00 [queued]>
[2023-12-21T16:52:56.737+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_ChestScan_with_mode scheduled__2023-12-01T00:00:00+00:00 [queued]>
[2023-12-21T16:52:56.745+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 6
[2023-12-21T16:52:56.787+0000] {taskinstance.py:1382} INFO - Executing <Task(PostgresOperator): impute_ChestScan_with_mode> on 2023-12-01 00:00:00+00:00
[2023-12-21T16:52:56.798+0000] {standard_task_runner.py:57} INFO - Started process 917 to run task
[2023-12-21T16:52:56.809+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings', 'impute_ChestScan_with_mode', 'scheduled__2023-12-01T00:00:00+00:00', '--job-id', '41', '--raw', '--subdir', 'DAGS_FOLDER/Indicators_of_Heart_Disease_2022.py', '--cfg-path', '/tmp/tmp75hxc_u0']
[2023-12-21T16:52:56.820+0000] {standard_task_runner.py:85} INFO - Job 41: Subtask impute_ChestScan_with_mode
[2023-12-21T16:52:57.040+0000] {task_command.py:416} INFO - Running <TaskInstance: Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings.impute_ChestScan_with_mode scheduled__2023-12-01T00:00:00+00:00 [running]> on host 8e01071a5eb3
[2023-12-21T16:52:57.224+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='dunnbc22' AIRFLOW_CTX_DAG_ID='Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings' AIRFLOW_CTX_TASK_ID='impute_ChestScan_with_mode' AIRFLOW_CTX_EXECUTION_DATE='2023-12-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-12-01T00:00:00+00:00'
[2023-12-21T16:52:57.230+0000] {sql.py:274} INFO - Executing: -- params: table_name, column_name

UPDATE indicators_of_heart_disease_2022_pipe
SET ChestScan = (
    SELECT ChestScan 
    FROM indicators_of_heart_disease_2022_pipe
    WHERE ChestScan IS NOT NULL
    GROUP BY ChestScan
    ORDER BY COUNT(*) DESC
    LIMIT 1
)
WHERE ChestScan IS NULL;
[2023-12-21T16:52:57.252+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2023-12-21T16:52:57.277+0000] {base.py:73} INFO - Using connection ID 'postgres_conn' for task execution.
[2023-12-21T16:52:57.300+0000] {sql.py:418} INFO - Running statement: -- params: table_name, column_name

UPDATE indicators_of_heart_disease_2022_pipe
SET ChestScan = (
    SELECT ChestScan 
    FROM indicators_of_heart_disease_2022_pipe
    WHERE ChestScan IS NOT NULL
    GROUP BY ChestScan
    ORDER BY COUNT(*) DESC
    LIMIT 1
)
WHERE ChestScan IS NULL;, parameters: None
[2023-12-21T16:52:58.539+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 280, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 423, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.DeadlockDetected: deadlock detected
DETAIL:  Process 74146 waits for ShareLock on transaction 1141751; blocked by process 74143.
Process 74143 waits for ShareLock on transaction 1141745; blocked by process 74137.
Process 74137 waits for ShareLock on transaction 1141748; blocked by process 74140.
Process 74140 waits for ShareLock on transaction 1141749; blocked by process 74141.
Process 74141 waits for ShareLock on transaction 1141753; blocked by process 74145.
Process 74145 waits for ShareLock on transaction 1141755; blocked by process 74147.
Process 74147 waits for ShareLock on transaction 1141750; blocked by process 74142.
Process 74142 waits for ShareLock on transaction 1141758; blocked by process 74151.
Process 74151 waits for ShareLock on transaction 1141754; blocked by process 74146.
HINT:  See server log for query details.
CONTEXT:  while rechecking updated tuple (2128,22) in relation "indicators_of_heart_disease_2022_pipe"

[2023-12-21T16:52:58.548+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=Indicators_of_Heart_Disease_2022-v01.impute_missing_values_strings, task_id=impute_ChestScan_with_mode, execution_date=20231201T000000, start_date=20231221T165256, end_date=20231221T165258
[2023-12-21T16:52:58.560+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 41 for task impute_ChestScan_with_mode (deadlock detected
DETAIL:  Process 74146 waits for ShareLock on transaction 1141751; blocked by process 74143.
Process 74143 waits for ShareLock on transaction 1141745; blocked by process 74137.
Process 74137 waits for ShareLock on transaction 1141748; blocked by process 74140.
Process 74140 waits for ShareLock on transaction 1141749; blocked by process 74141.
Process 74141 waits for ShareLock on transaction 1141753; blocked by process 74145.
Process 74145 waits for ShareLock on transaction 1141755; blocked by process 74147.
Process 74147 waits for ShareLock on transaction 1141750; blocked by process 74142.
Process 74142 waits for ShareLock on transaction 1141758; blocked by process 74151.
Process 74151 waits for ShareLock on transaction 1141754; blocked by process 74146.
HINT:  See server log for query details.
CONTEXT:  while rechecking updated tuple (2128,22) in relation "indicators_of_heart_disease_2022_pipe"
; 917)
[2023-12-21T16:52:58.584+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-12-21T16:52:58.653+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
